src/Groups/GAPGroupsLibrary.jl

```diff
@@ -0,0 +1,476 @@
+   elseif f == iscyclic return Bool, GAP.Globals.IsCyclic
+   elseif f == isnilpotent return Bool, GAP.Globals.IsNilpotentGroup
+   elseif f == isperfect return Bool, GAP.Globals.IsPerfectGroup
+   elseif f == issimple return Bool, GAP.Globals.IsSimpleGroup
+   elseif f == issolvable return Bool, GAP.Globals.IsSolvableGroup
+   elseif f == issupersolvable return Bool, GAP.Globals.IsSupersolvableGroup
+   elseif f == istransitive return Bool, GAP.Globals.IsTransitive
+   elseif f == number_moved_points || f == degree
+      if isapg
+         return Union{Int, AbstractVector{Int}}, GAP.Globals.NrMovedPoints
+      else
+         throw(ArgumentError("Function not supported"))
+      end
+   elseif f == order return Union{Int, AbstractVector{Int}}, GAP.Globals.Size
+   else throw(ArgumentError("Function not supported"))
+   end
+end
+
+function CheckValidType(L; isapg=false)    # isapg = Is a Permutation Group
+   isargument = false
+   expected_type = Any
+   arguments_to_add = 0                          # times the argument of a Boolean function is not specified
+   
+   for i in 1:length(L)
+      if isargument
+         if expected_type == Bool
+            if typeof(L[i]) <: Function
+               expected_type = find_index_function(L[i], isapg)[1]
+               arguments_to_add += 1
+            elseif typeof(L[i])==Bool
+               isargument = false
+            else
+               return false
+            end
+         else
+            if typeof(L[i]) <: expected_type
+               isargument = false
+            else
+               return false
+            end
+         end
+      else
+         if !(typeof(L[i]) <: Function)
+            return false
+         else
+            isargument = true
+            expected_type = find_index_function(L[i], isapg)[1]
+         end
+      end
+   end
+
+   if isargument
+      if expected_type == Bool
+         arguments_to_add += 1
+         return true, arguments_to_add
+      else
+         return false
+      end
+   else
+      return true, arguments_to_add
+   end
+end
+
+#=
+StoredValues = [ [ isabelian, Bool, GAP.Globals.IsAbelian ],
+   [ isalmostsimple, Bool, GAP.Globals.IsAlmostSimpleGroup ],
+   [ iscyclic, Bool, GAP.Globals.IsCyclic ],
+   [ isnilpotent, Bool, GAP.Globals.IsNilpotentGroup ],
+   [ isperfect, Bool, GAP.Globals.IsPerfectGroup ],
+   [ issimple, Bool, GAP.Globals.IsSimpleGroup ],
+   [ issolvable, Bool, GAP.Globals.IsSolvableGroup ],
+   [ issupersolvable, Bool, GAP.Globals.IsSupersolvableGroup ],
+   [ istransitive, Bool, GAP.Globals.IsTransitive ],
+   [ degree, Union{Int, AbstractVector{Int}}, GAP.Globals.Degree ]
+   [ order, Union{Int, AbstractVector{Int}}, GAP.Globals.Size ]
+]
+
+function find_index_function(f)
+   for i in 1::length(StoredValues)
+      if StoredValues[i][1]==f
+         return true, StoredValues[i]
+         break
+      end
+   end
+
+   return false, Nothing
+end
+
```

Comment:
This code and more below it, is commented out. Why is it here? Do you still plan to use it, or is it a leftover and can be removed?